client/dashboard/sites/settings-subscription-gifting/index.tsx

```diff
@@ -48,8 +63,25 @@ export default function SubscriptionGiftingSettings( { siteSlug }: { siteSlug: s
 		throw notFound();
 	}
 
-	const handleSubmit = ( edits: Partial< SiteSettings > ) => {
-		return mutation.mutate( edits );
+	const isDirty = Object.entries( formData ).some(
+		( [ key, value ] ) => data[ key as keyof SiteSettings ] !== value
+	);
+
+	const { isPending } = mutation;
+
+	const handleSubmit = ( e: React.FormEvent ) => {
```

Comment:
Maybe it's too soon, but settings form are starting to shape in a similar way:

 - state
 - defining site settings fields
 - submit function
 - isDirty check
 - notices

It could be that it's too soon to abstract but maybe there's a shareable light component here. The other thing is that I wonder if the "fields" definition should be entirely separate from the components. In Gutenberg, we started a `@wordpress/fields` package for that, I wonder if ultimately here, we could have

 - fields/site for site fields
 - fields/site-settings for site settings fields
 - fields/domain for domain fields

....

cc @Automattic/lego just something to think about just in case.

Reply:
Some field's options depend on some value of the site. See e.g.

https://github.com/Automattic/wp-calypso/pull/103668/files#diff-5aac1afa06bffa37958dc1c949fd604c9bf14fdcd8712bb09b65f1038aa1ec11R47-R53

Reply:
@fushar potentially the "elements" could be injected at runtime? it's not mandatory to define the field itself.